web/vtadmin/src/components/routes/tablets/InfoDialog.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useEffect, useState } from 'react';
+import { QueryHook, Options } from '../../../hooks/api';
+import { Transition } from '@headlessui/react';
+import Dialog from '../../dialog/Dialog';
+import { Icon, Icons } from '../../Icon';
+
+export interface BaseInfoDialogProps {
+    loadingTitle?: string;
+    loadingDescription: string;
+    successTitle?: string;
+    successDescription: string;
+    errorTitle?: string;
+    errorDescription: string;
+    useHook: QueryHook;
```

Comment:
I really like the way you abstracted the `useHook` prop. :chef-kiss: I think the only downside is that we lose the typesafe mappings between `useHook`, `options`, and `params` because of the `any`s in `QueryHook`.

It looks like we can maybe get around it by slapping a closure on the whole `useHook` query + all its arguments before passing it as a prop (sorry for the gross diff):

```diff
diff --git a/web/vtadmin/src/components/routes/tablets/InfoDialog.tsx b/web/vtadmin/src/components/routes/tablets/InfoDialog.tsx
index 2bf401e555..c4f9e8ddd5 100644
--- a/web/vtadmin/src/components/routes/tablets/InfoDialog.tsx
+++ b/web/vtadmin/src/components/routes/tablets/InfoDialog.tsx
@@ -1,8 +1,8 @@
 import React, { useEffect, useState } from 'react';
-import { QueryHook, Options } from '../../../hooks/api';
 import { Transition } from '@headlessui/react';
 import Dialog from '../../dialog/Dialog';
 import { Icon, Icons } from '../../Icon';
+import { UseQueryResult } from 'react-query';
 
 export interface BaseInfoDialogProps {
     loadingTitle?: string;
@@ -11,9 +11,7 @@ export interface BaseInfoDialogProps {
     successDescription: string;
     errorTitle?: string;
     errorDescription: string;
-    useHook: QueryHook;
-    params: Record<string, any>;
-    options: Options;
+    useHook: () => UseQueryResult<any, Error>;
 }
 
 interface InfoDialogProps extends BaseInfoDialogProps {
@@ -29,12 +27,10 @@ const InfoDialog: React.FC<InfoDialogProps> = ({
     errorTitle,
     errorDescription,
     useHook,
-    params,
-    options,
     isOpen,
     onClose,
 }) => {
-    const { data, error, isLoading, refetch } = useHook(params, options || {});
+    const { data, error, isLoading, refetch } = useHook();
 
     // Animate loading briefly in case useHook is very fast
     // to give UX sense of work being done
diff --git a/web/vtadmin/src/components/routes/tablets/InfoDropdown.tsx b/web/vtadmin/src/components/routes/tablets/InfoDropdown.tsx
index 17540c9e02..dd7616b539 100644
--- a/web/vtadmin/src/components/routes/tablets/InfoDropdown.tsx
+++ b/web/vtadmin/src/components/routes/tablets/InfoDropdown.tsx
@@ -21,27 +21,21 @@ const InfoDropdown: React.FC<InfoDropdownProps> = ({ alias, clusterID }) => {
 
     const dialogConfigs: Record<string, BaseInfoDialogProps> = {
         ping: {
-            useHook: usePingTablet,
-            params: { alias, clusterID },
-            options: { enabled: false },
+            useHook: () => usePingTablet({ alias, clusterID }, { enabled: false }),
             successDescription: `Successfully reached tablet ${alias} via RPC.`,
             errorDescription: `There was an issue pinging tablet ${alias}`,
             loadingTitle: `Pinging tablet ${alias}`,
             loadingDescription: 'Checking to see if tablet is reachable via RPC...',
         },
         refresh: {
-            useHook: useRefreshState,
-            params: { alias, clusterID },
-            options: { enabled: false },
+            useHook: () => useRefreshState({ alias, clusterID }, { enabled: false }),
             successDescription: `Successfully refreshed tablet ${alias}.`,
             errorDescription: `There was an issue refreshing tablet`,
             loadingTitle: `Refreshing tablet ${alias}`,
             loadingDescription: 'Refreshing tablet record on tablet...',
         },
         healthcheck: {
-            useHook: useHealthCheck,
-            params: { alias, clusterID },
-            options: { enabled: false },
+            useHook: () => useHealthCheck({ alias, clusterID }, { enabled: false }),
             successDescription: `Tablet ${alias} looks healthy.`,
             errorDescription: `There was an issue running a health check on the tablet`,
             loadingTitle: `Running health check`,
diff --git a/web/vtadmin/src/hooks/api.ts b/web/vtadmin/src/hooks/api.ts
index f88b9b084e..b354cb6c3c 100644
--- a/web/vtadmin/src/hooks/api.ts
+++ b/web/vtadmin/src/hooks/api.ts
@@ -276,6 +276,3 @@ export const useWorkflow = (
         ...options,
     });
 };
-
-export type QueryHook = (params: any, options?: any) => UseQueryResult<any, Error>;
-export type Options = UseQueryOptions<ResponseType, Error> | undefined;
```

With this, something like this will (correctly) fail type-checking:

```
  const dialogConfigs: Record<string, BaseInfoDialogProps> = {
      ping: {
          useHook: () => usePingTablet({ alias: 123, clusterID }, { enabled: false }),
```

Reply:
That's a great idea! :-) I'll make the change.